jsdom-nodejs-env/src/main/scala/org/scalajs/jsenv/jsdomnodejs/JSDOMNodeJSEnv.scala

```diff
@@ -42,9 +43,18 @@ class JSDOMNodeJSEnv(config: JSDOMNodeJSEnv.Config) extends JSEnv {
   def startWithCom(input: Input, runConfig: RunConfig,
       onMessage: String => Unit): JSComRun = {
     JSDOMNodeJSEnv.validator.validate(runConfig)
+    val scripts = validateInput(input)
     ComRun.start(runConfig, onMessage) { comLoader =>
-      val files = initFiles ::: (comLoader :: codeWithJSDOMContext(input))
-      internalStart(files, runConfig)
+      internalStart(comLoader :: codeWithJSDOMContext(scripts), runConfig)
+    }
+  }
+
+  private def validateInput(input: Input): List[VirtualBinaryFile] = {
```

Comment:
I don't fully understand why `scriptFiles` get's replaced with this (equivalent) method (also `extractInput` or something would seem to be a more appropriate name).

Reply:
also I'm unclear on why it get's moved out of `codeWithJSDOMContext`, it seems logically clearer there (`ComRun.start` is evaluated eagerly, right?)

Reply:
`ComRun.start` is evaluated eagerly, yes, but all non-fatal exceptions will be wrapped in a failed `ComRun`. But `UnsupportedInputException` is supposed to be thrown outside of the `ComRun`. So that's why moved out of `codeWithJSDOMContext`. And it's renamed to make it very clear that it validates the input at that time, potentially throwing the `UnsupportedInputException` at that time, and not later.

I also hesitated with `extractInput`, so that's another possibility.

Reply:
Ok. So in this case it's fine. Let's leave it then.